### PR TITLE
[2.3.0] Dynamic priority fee support

### PR DESCRIPTION
**Pull request by worlddlckgh**

### CODE REVIEW

**Review by chiew256 — APPROVED** [review]

File: src/contexts/PrioritizationFeeContextProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          : null;
+
+        let priceMicroLamports = 0;
+
+        const marketAndRpcReference = Math.max(options.referenceFee || 0, minimumFee || 0);
+
+        priceMicroLamports = Math.round(
+          priorityMode === 'EXACT'
+            ? userMaxPriorityFee
+            : Math.min(userMaxPriorityFee, Math.min(userMaxPriorityFee, marketAndRpcReference)),
```
[the remark this answers is not in the document]
Comment (worlddlckgh) [reply src/contexts/PrioritizationFeeContextProvider.tsx:109]: good catch

File: src/contexts/SwapContext.tsx [hunk 2]
```diff
@@ -371,7 +336,13 @@ export const SwapContextProvider: FC<{
         body: JSON.stringify({
           quoteResponse: quoteResponseMeta.original,
           userPublicKey: walletPublicKey,
+          /**
+           * Note: This is not optimal `computeUnitLimit` as it's not calculated based on the transaction `instructions`
+           * This is just a placeholder value to make the API call, we will replace it with the optimal value later,
+           * by simulate the transaction and get the optimal value.
+           */
           computeUnitPriceMicroLamports,
```
[the remark this answers is not in the document]
Comment (worlddlckgh) [reply src/contexts/SwapContext.tsx]: removed, thanks

File: src/contexts/PrioritizationFeeContextProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  priorityFeeLamports: 0,
+
+  // method
+  setPriorityFee: () => {},
+  setPriorityMode: () => {},
+  setPriorityLevel: () => {},
+  modifyComputeUnitPriceAndLimit: async () => {},
+});
+
+export function PrioritizationFeeContextProvider({ children }: { children: React.ReactNode }) {
```
Comment (wkitsin) [inline src/contexts/PrioritizationFeeContextProvider.tsx:67]: small issue This should be named `PrioritizationFeeProvider`

**Review by wkitsin — COMMENTED** [review]
LGTM

**Pull request merged**